core/domain/user_services.py

```diff
@@ -405,3 +407,120 @@ def get_email_preferences(user_id):
             if email_preferences_model is None
             else email_preferences_model.site_updates)
     }
+
+
+class UserContributions(object):
+    """Value object representing a user's contributions."""
+    def __init__(
+            self, user_id, created_exploration_ids, edited_exploration_ids):
+        self.user_id = user_id
+        self.created_exploration_ids = created_exploration_ids
+        self.edited_exploration_ids = edited_exploration_ids
+
+    def validate(self):
+        if not isinstance(self.user_id, basestring):
+            raise utils.ValidationError(
+                'Expected user_id to be a string, received %s' % self.user_id)
+        if not isinstance(self.created_exploration_ids, list):
+            raise utils.ValidationError(
+                'Expected created_exploration_ids to be a list, received %s' 
+            % self.created_exploration_ids)
+        if not isinstance(self.edited_exploration_ids, list):
+            raise utils.ValidationError(
+                'Expected edited_exploration_ids to be a list, received %s' 
+            % self.edited_exploration_ids)
+        if not self.user_id:
```

Comment:
Please put all the checks for user_id at the top, all the checks for created_exploration_ids after that, and all the checks for edited_exploration_ids after that. Separate the blocks with newlines to give the code structure -- right now, everything is a bit jumbled.